### PR TITLE
Generate markdown changelog

### PR DESCRIPTION
**Pull request by jaseemabid**
This patch add a simple API to generate markdown based changelog for a
particular sass-openshift context. There are still some rough edges, but I thought I'll share what I have right now before proceeding further. This definitely needs some tests at least. 

Fixes #1420. 

Here is a sample run

```
In [360]: from saasherder.changelog import Changelog
In [361]: c = Changelog()
In [362]: c.main('dsaas', 'master', 'master~4')

04d18c56 Component test improvements (#2234)
d9bfcc70 fix(deployments): resource-card unsubscribes on destroy
8c22ac04 chore(docs): remove a dulicate issue_template
a65b98c0 chore(docs): add issue template to redirect to OSIO repo

959f75f Some cleanups suggested by golint (#1778)
a19c642 Rework controller link tests (#1769)
dc5c4b9 Explain what WIT is (#1776)
7278954 Add "workItemLinks" relationship on work item response (#1775)

8453fc8 Merge pull request #479 from fabric8-services/versionUpdate28b384c4-fb33-4ed6-9090-e4ff5a251acf
d1dea73 Update fabric8-tenant-che version to 2.0.63
839c30c Reverse lookup tenant by cluster/namespace (openshiftio/openshiftio#1389)
bdbb91e Merge branch 'master' into IssueOSIO1389_reverse_lookup_tenant
611824e Merge pull request #478 from fabric8-services/versionUpdate06df89a5-2c6d-40b4-bb5d-f2820f4b5962
088bb05 Changing number of parameters in tests from 3 to 5 due to che template update
8b433f3 Update fabric8-tenant-che version to 2.0.62
05df58c Merge branch 'master' into IssueOSIO1389_reverse_lookup_tenant
2be1a3e Add `requestId` and `identityId` to the template variables for the Che migration job POD
cccdb41 Revert "Also modifies the test file"
8bde08c Also modifies the test file
b1352fa Apply review comments and fixing test failures :/
96f16b8 Reverse lookup tenant by cluster/namespace (openshiftio/openshiftio#1389)
4422022 Add missing import
b076a55 Correctly manage the error case when the `sub` claim is not found
b375326 Fix formatting error returned by gofmt
3e145a7 fix formatting
82ea019 syntax error
80f3081 Fix errors
252b900 Add `requestId` and `identityId` to the template variables ...
```

### CODE REVIEW

**Comment by jaseemabid** [comment]
PS: This patch is built on top of my previous unmerged PR. I'll rebase after review. Ignore the first 2 patches in the series.

**Comment by kbsingh** [comment]
xref task from https://github.com/openshiftio/openshift.io/issues/1420

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.PHONY: build-test-container
```
Comment (maxandersen) [inline Makefile]: when I run these I get errors like:

```
tests/test_saasherder.py:6: in <module>
    from saasherder import SaasHerder
saasherder/saasherder.py:10: in <module>
    from .config import SaasConfig
E   ValueError: Attempted relative import in non-package
______________________________________________________________________ ERROR collecting tests/test_template.py ______________________________________________________________________
tests/test_template.py:6: in <module>
    from saasherder import SaasHerder
saasherder/saasherder.py:10: in <module>
    from .config import SaasConfig
E   ValueError: Attempted relative import in non-package
```

**Comment by maxandersen** [comment]
trying to run it locally - what commands are you running to reproduce your output ?

File: saasherder/changelog.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        names = self.service_names(context)
+        dsls = [self.definitions(context, name) for name in names]
+        return list(chain(*dsls))
+
+    @staticmethod
+    def diff(s1, s2):
+        """Return the difference between 2 service lists
+
+        This is not a commutative relationship. We are looking for services in
+        first list which differ from that in second list. a ⊕ b ≠ b ⊕ a.
```
Comment (maxandersen) [inline saasherder/changelog.py]: these special characters result in:

" File "../saasherder/saasherder/changelog.py", line 47
SyntaxError: Non-ASCII character '\xe2' in file ../saasherder/saasherder/changelog.py on line 48, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details"

File: saasherder/changelog.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+        # Checkout to previous version and get services
+        self.run("git checkout {}".format(end))
+        previous = self.services(context)
+
+        self.run("git checkout {}".format(start))
+        now = self.services(context)
+
+        # Go back to where we were
+        self.run("git checkout master")
```
Comment (maxandersen) [inline saasherder/changelog.py]: why do you need to be specifcally on `master` ?

File: saasherder/changelog.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        return [{
+            'name': service['name'],
+            'url': service['url'],
+            'new': service.get('hash', 'master'),  # NOTE: null here is silly
+            'old': old(service)
+        } for service in useful]
+
+    @staticmethod
+    def run(cmd):
+        return subprocess.run(cmd, stdout=subprocess.PIPE,
```
Comment (maxandersen) [inline saasherder/changelog.py]: `.run(...)` only exist for python3 - is that expected ?

Comment (maxandersen) [inline saasherder/changelog.py]: using this on `saas-openshift` repo makes it just sit there for a while - worth adding a `print ('Running ' + cmd)` to see progres.

File: saasherder/changelog.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        self.run("git checkout master")
+
+        changed = self.diff(previous, now)
+
+        # Fetch all the services that changed
+        for service in changed:
+            self.checkout(service, service['new'])
+
+        # Get the changelog for each service as a `(name, [messages])` tuple
+        changelog = [(service['name'], self.log(service, start, end)) for
```
Comment (maxandersen) [inline saasherder/changelog.py]: `start` and `end` here are not actually the relevant hash for what changed in the services. Thus it is actually not showing the true diff.

Should use the start/end from the service info, something like:

` changelog = [(service['name'], self.log(service, service['old'], service['new'])) for`

File: saasherder/changelog.py (hunk 2 above)
Comment (maxandersen) [reply saasherder/changelog.py]: this was because I was running with `python` not `python3`

File: saasherder/changelog.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        return self.run(cmd).stdout.decode('utf-8')
+
+    @staticmethod
+    def markdown(changelog, start, end):
+        """Pretty print the change log
+
+        The incoming object is a list of `(name, messages)` tuples.
+        """
+
+        header = "# Changelog \n\n## From version {} to {}".format(start, end)
```
Comment (maxandersen) [inline saasherder/changelog.py]: missing a \n

File: Makefile (hunk 1 above)
Comment (jaseemabid) [reply Makefile]: You are probably on Python 2.  I wasn't sure which one to support but both container images in the repo gets built with Py3 so went ahead with it.

File: saasherder/changelog.py (hunk 3 above)
Comment (jaseemabid) [reply saasherder/changelog.py]: I can get the reference before; save it and go back to the same place.

File: saasherder/changelog.py (hunk 4 above)
Comment (jaseemabid) [reply saasherder/changelog.py]: Hi @vpavlin, do we have to support 2? I'm not sure. Since we run this ourselves in constrained environments, I guess we have some freedom here.

Comment (jaseemabid) [reply saasherder/changelog.py]: @maxandersen Yes, we need some more logging in general. I'll add that.

File: saasherder/changelog.py (hunk 5 above)
Comment (jaseemabid) [reply saasherder/changelog.py]: @maxandersen +1. My bad :/ 👍

**Comment by vpavlin** [comment]
[test]

File: saasherder/cli.py [hunk 7]
```diff
@@ -2,8 +2,8 @@
 
 import argparse
 
-from saasherder import SaasHerder
-from config import SaasConfig
+from .saasherder import SaasHerder
```
Comment (vpavlin) [inline saasherder/cli.py]: This (and other changes like this one) should not be part of this PR - it is not relevant for the change, please push it as a separate PR

File: Makefile (hunk 1 above)
Comment (vpavlin) [reply Makefile]: I don't mind using Python3, but it should be introduced in a separate PR and you'll need to fix the Dockerfile to actually use Python3 - otherwise tests fail https://ci.centos.org/view/Devtools/job/devtools-saasherder/4/console

File: saasherder/changelog.py (hunk 4 above)
Comment (vpavlin) [reply saasherder/changelog.py]: As mentioned elsewhere - you can add support of Py3, but do it in separate repo and you will need to make it work for 3 in the container - keep in mind that this is out tool to promote to prod for OpenShift.io, we should not break it;)

File: saasherder/changelog.py (hunk 3 above)
Comment (vpavlin) [reply saasherder/changelog.py]: Theoretically you don't go where you were...you go to master, which might be different than where you were:)

File: saasherder/changelog.py [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+        body = ["## {} \n{} \n".format(name, messages)
+                for (name, messages) in changelog]
+
+        return header + ''.join(body)
```
Comment (vpavlin) [inline saasherder/changelog.py]: you could just put everything in one list and `'\n'.join(body)`

**Comment by vpavlin** [comment]
I'll do a thorough review (and actually run it) over the weekend, but my first observation is that it does not use saasherder code, which is not right, because then we duplicate what we already have - the reason I suggested it should be a subcommand of saasherder was to prevent this.

Code from here https://github.com/openshiftio/saasherder/pull/41/files#diff-5f6b41a0351b3c296fa6172d648dfb4eR11 till line 40 should be in saasherder already and should be reused (if something is missing, I'd suggest to add it as a method of SaasHerder class)

@jaseemabid What do you think?

**Comment by maxandersen** [comment]
btw. Here is a proposal on how the changelog could look like. Added links to see more details for github + dates and authors.

## [saas-openshiftio](https://github.com/openshiftio/saas-openshiftio): [master..master~4](https://github.com/openshiftio/saas-openshiftio/compare/master..master)

### [fabric8-tenant](https://github.com/fabric8-services/fabric8-tenant): [8453fc8..96f16b8](https://github.com/fabric8-services/fabric8-tenant/compare/8453fc8d3d725c96e278c6d2218eaf9a2192fed8?expand=1) 

- [8453fc8](https://github.com/fabric8-services/fabric8-tenant/commit/8453fc8)	fabric8-cd	2017-11-24	Merge pull request #479 from fabric8-services/versionUpdate28b384c4-fb33-4ed6-9090-e4ff5a251acf
- [d1dea73](https://github.com/fabric8-services/fabric8-tenant/commit/d1dea73)	fabric8-cd	2017-11-24	Update fabric8-tenant-che version to 2.0.63
- [839c30c](https://github.com/fabric8-services/fabric8-tenant/commit/839c30c)	Xavier Coulon	2017-11-23	Reverse lookup tenant by cluster/namespace (openshiftio/openshiftio#1389)
- [bdbb91e](https://github.com/fabric8-services/fabric8-tenant/commit/bdbb91e)	Xavier Coulon	2017-11-23	Merge branch 'master' into IssueOSIO1389_reverse_lookup_tenant
- [611824e](https://github.com/fabric8-services/fabric8-tenant/commit/611824e)	fabric8-cd	2017-11-22	Merge pull request #478 from fabric8-services/versionUpdate06df89a5-2c6d-40b4-bb5d-f2820f4b5962
- [088bb05](https://github.com/fabric8-services/fabric8-tenant/commit/088bb05)	Ilya Buziuk	2017-11-22	Changing number of parameters in tests from 3 to 5 due to che template update
- [8b433f3](https://github.com/fabric8-services/fabric8-tenant/commit/8b433f3)	fabric8-cd	2017-11-22	Update fabric8-tenant-che version to 2.0.62
- [05df58c](https://github.com/fabric8-services/fabric8-tenant/commit/05df58c)	Xavier Coulon	2017-11-22	Merge branch 'master' into IssueOSIO1389_reverse_lookup_tenant
- [b1352fa](https://github.com/fabric8-services/fabric8-tenant/commit/b1352fa)	Xavier Coulon	2017-11-22	Apply review comments and fixing test failures :/
- [96f16b8](https://github.com/fabric8-services/fabric8-tenant/commit/96f16b8)	Xavier Coulon	2017-11-22	Reverse lookup tenant by cluster/namespace (openshiftio/openshiftio#1389)

**Comment by jaseemabid** [comment]
@vpavlin Some of the changes were introduced because I assumed Py3. My bad. Like changing import styles. Reverting all that. About reusing existing code, I can do that too.

**Comment by jaseemabid** [comment]
@maxandersen I'm fine with any format for the changelog. I can go with what you suggested. I just had to start somewhere.

File: Makefile (hunk 1 above)
Comment (jaseemabid) [reply Makefile]: @maxandersen @vpavlin Switching back to Py2.

File: saasherder/changelog.py (hunk 2 above)
Comment (jaseemabid) [reply saasherder/changelog.py]: Ah! Py 3 didn't give me a warning. I'll add the encoding header.

**Comment by jaseemabid** [comment]
@maxandersen I agree that links to commits will help but other information like date and author name looks like noise. Not a very strong opinion though. Just 0.02$.

**Comment by jaseemabid** [comment]
# [saas-openshift.io](https://github.com/openshiftio/saas-openshiftio): [master..master~4](https://github.com/openshiftio/saas-openshiftio/compare/master..master~4) 

## fabric8-ui 
[6da6fc22](https://github.com/fabric8-ui/fabric8-ui/commit/6da6fc22b5d66c480cdf3d61cf1f5437cab9a8c6) fix(color): Update colors on _home banner
[46c8cf12](https://github.com/fabric8-ui/fabric8-ui/commit/46c8cf12b6382343dcdf1f8f6f918d192f9812ba) fix(toolbar): Set position to top, remove media query
[16cbcc2c](https://github.com/fabric8-ui/fabric8-ui/commit/16cbcc2cc0924060d71a4f167e467b33dff55f20) fix(tooltip): fix alignment and update per PR review
[d58355e6](https://github.com/fabric8-ui/fabric8-ui/commit/d58355e67ac9e91b81d5e4d7b024ddca7eefc80b) fix(tooltip): Fix tooltip alignment in toolbar
[7feb25d5](https://github.com/fabric8-ui/fabric8-ui/commit/7feb25d5750e978456ca1237c866321852d6acf1) refactor(deployments): rename AppsComponent to DeploymentsComponent
[1ed4a709](https://github.com/fabric8-ui/fabric8-ui/commit/1ed4a7094036c9c10e0c4430947e2312805f8803) refactor(deployments): rename apps.component files to deployments.component
 
## fabric8-wit 
[7278954](https://github.com/fabric8-services/fabric8-wit/commit/72789544a3f75cb6a805531ec09ce282b649b445) Add "workItemLinks" relationship on work item response (#1775)
[5fae1e9](https://github.com/fabric8-services/fabric8-wit/commit/5fae1e99ffa6f04e475961cc11b74edb12725d19) Filter by title (#1748)
[1f2f6b7](https://github.com/fabric8-services/fabric8-wit/commit/1f2f6b7c065422e93ef7fe3faa008e24df7869ca) Enabling optional fields in User/Identity data-model to accept empty strings
[4f6c79d](https://github.com/fabric8-services/fabric8-wit/commit/4f6c79d9efe33c96af18dc6bd6b8845384eaa4eb) UUID agnostic golden file comparison replaces real UUIDs during update (#1774)
[5ee5e5d](https://github.com/fabric8-services/fabric8-wit/commit/5ee5e5d58d588a94b462b0a00203ab9b4bacf847) Improve TestSearchController.TestUpdateWorkItem() test (#1773)
[8ceb1df](https://github.com/fabric8-services/fabric8-wit/commit/8ceb1df099f248844deed7ebea2c450eaf341b9a) Local core & planner OpenShift development using Kedge (#1741)
[2ec6900](https://github.com/fabric8-services/fabric8-wit/commit/2ec690009776a1f952f3753add0fa1c1e55a0e69) Parameterised the isServiceAccount(..) method (#1767)
 
## fabric8-tenant 
[ba345e5](https://github.com/fabric8-services/fabric8-tenant/commit/ba345e5f13c1791732ef4c22f6a4c019958fac77) Merge pull request #472 from xcoulon/Issue471_test_fixture
[781e211](https://github.com/fabric8-services/fabric8-tenant/commit/781e211c9af50cc6de079a5a7a743e874df30115) Include TextFixture lib to test the DB services (#471)

**Comment by jaseemabid** [comment]
Update, we have something like this now. 

# [saas-openshift.io](https://github.com/openshiftio/saas-openshiftio): [master..master~4](https://github.com/openshiftio/saas-openshiftio/compare/master~4...master) 

## [fabric8-ui](https://github.com/fabric8-ui/fabric8-ui/): [6da6fc2..562e00b](https://github.com/fabric8-ui/fabric8-ui//compare/562e00bb528c3fb285d16dcd5602c62611e2b4cc...6da6fc22b5d66c480cdf3d61cf1f5437cab9a8c6)
[6da6fc22](https://github.com/fabric8-ui/fabric8-ui/commit/6da6fc22b5d66c480cdf3d61cf1f5437cab9a8c6) fix(color): Update colors on _home banner
[46c8cf12](https://github.com/fabric8-ui/fabric8-ui/commit/46c8cf12b6382343dcdf1f8f6f918d192f9812ba) fix(toolbar): Set position to top, remove media query
[16cbcc2c](https://github.com/fabric8-ui/fabric8-ui/commit/16cbcc2cc0924060d71a4f167e467b33dff55f20) fix(tooltip): fix alignment and update per PR review
[d58355e6](https://github.com/fabric8-ui/fabric8-ui/commit/d58355e67ac9e91b81d5e4d7b024ddca7eefc80b) fix(tooltip): Fix tooltip alignment in toolbar
[7feb25d5](https://github.com/fabric8-ui/fabric8-ui/commit/7feb25d5750e978456ca1237c866321852d6acf1) refactor(deployments): rename AppsComponent to DeploymentsComponent
[1ed4a709](https://github.com/fabric8-ui/fabric8-ui/commit/1ed4a7094036c9c10e0c4430947e2312805f8803) refactor(deployments): rename apps.component files to deployments.component

## [fabric8-wit](https://github.com/fabric8-services/fabric8-wit/): [7278954..8a65bcb](https://github.com/fabric8-services/fabric8-wit//compare/8a65bcbc7bb2efaaa8c56337d076a3e9e0b19179...72789544a3f75cb6a805531ec09ce282b649b445)
[7278954](https://github.com/fabric8-services/fabric8-wit/commit/72789544a3f75cb6a805531ec09ce282b649b445) Add "workItemLinks" relationship on work item response (#1775)
[5fae1e9](https://github.com/fabric8-services/fabric8-wit/commit/5fae1e99ffa6f04e475961cc11b74edb12725d19) Filter by title (#1748)
[1f2f6b7](https://github.com/fabric8-services/fabric8-wit/commit/1f2f6b7c065422e93ef7fe3faa008e24df7869ca) Enabling optional fields in User/Identity data-model to accept empty strings
[4f6c79d](https://github.com/fabric8-services/fabric8-wit/commit/4f6c79d9efe33c96af18dc6bd6b8845384eaa4eb) UUID agnostic golden file comparison replaces real UUIDs during update (#1774)
[5ee5e5d](https://github.com/fabric8-services/fabric8-wit/commit/5ee5e5d58d588a94b462b0a00203ab9b4bacf847) Improve TestSearchController.TestUpdateWorkItem() test (#1773)
[8ceb1df](https://github.com/fabric8-services/fabric8-wit/commit/8ceb1df099f248844deed7ebea2c450eaf341b9a) Local core & planner OpenShift development using Kedge (#1741)
[2ec6900](https://github.com/fabric8-services/fabric8-wit/commit/2ec690009776a1f952f3753add0fa1c1e55a0e69) Parameterised the isServiceAccount(..) method (#1767)

## [fabric8-tenant](https://github.com/fabric8-services/fabric8-tenant/): [ba345e5..1230eb3](https://github.com/fabric8-services/fabric8-tenant//compare/1230eb36da21800589a0aa979b2a1924d6c2c488...ba345e5f13c1791732ef4c22f6a4c019958fac77)
[ba345e5](https://github.com/fabric8-services/fabric8-tenant/commit/ba345e5f13c1791732ef4c22f6a4c019958fac77) Merge pull request #472 from xcoulon/Issue471_test_fixture
[781e211](https://github.com/fabric8-services/fabric8-tenant/commit/781e211c9af50cc6de079a5a7a743e874df30115) Include TextFixture lib to test the DB services (#471)

**Comment by jaseemabid** [comment]
@vpavlin I've removed some of the redundant code I added to changelog module and integrated it as a submodule. 

You can use it like ```saasherder changelog --context dsaas master master~4``` now.

**Comment by vpavlin** [comment]
[test]

**Comment by jaseemabid** [comment]
Rebasing this on top of the current master since we just merged #40.

**Comment by vpavlin** [comment]
[test]

File: saasherder/cli.py [hunk 9]
```diff
@@ -58,9 +58,15 @@ def main():
     subparser_template.add_argument("type", choices=["get-contexts"],
                                     help="Subcommands for config command")
 
+    subparser_changelog = subparsers.add_parser("changelog")
+    subparser_changelog.add_argument("--context", action="store")
```
Comment (vpavlin) [inline saasherder/cli.py:62]: Context is a global argument, you don't need it here

**Comment by vpavlin** [comment]
[test]

**Comment by jaseemabid** [comment]
Adding a last updated time as per @maxandersen's comments.

# [saas-openshift.io](https://github.com/openshiftio/saas-openshiftio): [master..master~4](https://github.com/openshiftio/saas-openshiftio/compare/master~4...master) 

## [fabric8-ui](https://github.com/fabric8-ui/fabric8-ui/): [6da6fc2..562e00b](https://github.com/fabric8-ui/fabric8-ui//compare/562e00bb528c3fb285d16dcd5602c62611e2b4cc...6da6fc22b5d66c480cdf3d61cf1f5437cab9a8c6)
Last updated at Mon Nov 20 23:37:48 2017 -0500 

[6da6fc22](https://github.com/fabric8-ui/fabric8-ui/commit/6da6fc22b5d66c480cdf3d61cf1f5437cab9a8c6) fix(color): Update colors on _home banner
[46c8cf12](https://github.com/fabric8-ui/fabric8-ui/commit/46c8cf12b6382343dcdf1f8f6f918d192f9812ba) fix(toolbar): Set position to top, remove media query
[16cbcc2c](https://github.com/fabric8-ui/fabric8-ui/commit/16cbcc2cc0924060d71a4f167e467b33dff55f20) fix(tooltip): fix alignment and update per PR review
[d58355e6](https://github.com/fabric8-ui/fabric8-ui/commit/d58355e67ac9e91b81d5e4d7b024ddca7eefc80b) fix(tooltip): Fix tooltip alignment in toolbar
[7feb25d5](https://github.com/fabric8-ui/fabric8-ui/commit/7feb25d5750e978456ca1237c866321852d6acf1) refactor(deployments): rename AppsComponent to DeploymentsComponent
[1ed4a709](https://github.com/fabric8-ui/fabric8-ui/commit/1ed4a7094036c9c10e0c4430947e2312805f8803) refactor(deployments): rename apps.component files to deployments.component

## [fabric8-wit](https://github.com/fabric8-services/fabric8-wit/): [7278954..8a65bcb](https://github.com/fabric8-services/fabric8-wit//compare/8a65bcbc7bb2efaaa8c56337d076a3e9e0b19179...72789544a3f75cb6a805531ec09ce282b649b445)
Last updated at Mon Nov 20 18:08:05 2017 +0100 

[7278954](https://github.com/fabric8-services/fabric8-wit/commit/72789544a3f75cb6a805531ec09ce282b649b445) Add "workItemLinks" relationship on work item response (#1775)
[5fae1e9](https://github.com/fabric8-services/fabric8-wit/commit/5fae1e99ffa6f04e475961cc11b74edb12725d19) Filter by title (#1748)
[1f2f6b7](https://github.com/fabric8-services/fabric8-wit/commit/1f2f6b7c065422e93ef7fe3faa008e24df7869ca) Enabling optional fields in User/Identity data-model to accept empty strings
[4f6c79d](https://github.com/fabric8-services/fabric8-wit/commit/4f6c79d9efe33c96af18dc6bd6b8845384eaa4eb) UUID agnostic golden file comparison replaces real UUIDs during update (#1774)
[5ee5e5d](https://github.com/fabric8-services/fabric8-wit/commit/5ee5e5d58d588a94b462b0a00203ab9b4bacf847) Improve TestSearchController.TestUpdateWorkItem() test (#1773)
[8ceb1df](https://github.com/fabric8-services/fabric8-wit/commit/8ceb1df099f248844deed7ebea2c450eaf341b9a) Local core & planner OpenShift development using Kedge (#1741)
[2ec6900](https://github.com/fabric8-services/fabric8-wit/commit/2ec690009776a1f952f3753add0fa1c1e55a0e69) Parameterised the isServiceAccount(..) method (#1767)

## [fabric8-tenant](https://github.com/fabric8-services/fabric8-tenant/): [ba345e5..1230eb3](https://github.com/fabric8-services/fabric8-tenant//compare/1230eb36da21800589a0aa979b2a1924d6c2c488...ba345e5f13c1791732ef4c22f6a4c019958fac77)
Last updated at Tue Nov 21 11:34:34 2017 +0100 

[ba345e5](https://github.com/fabric8-services/fabric8-tenant/commit/ba345e5f13c1791732ef4c22f6a4c019958fac77) Merge pull request #472 from xcoulon/Issue471_test_fixture
[781e211](https://github.com/fabric8-services/fabric8-tenant/commit/781e211c9af50cc6de079a5a7a743e874df30115) Include TextFixture lib to test the DB services (#471)

**Comment by vpavlin** [comment]
[test]
